README.md

```diff
@@ -162,27 +162,37 @@ const resolvers = {
 
 # Typescript
 
-As the library cannot know the structure of your user object you need to add this to the global scope by writing your own "somename.d.ts" you can adapt the following template with whatever user variables that you want have added to the PassportRequest:
+As the library cannot know the structure of your user object you need to add this to the global scope by writing your own "somename.d.ts" you can add how your context definition should look like:
 
 ```ts
-declare module "graphql-passport" {
-  interface PassportRequest {
-    user?: {
-      username: string;
-    };
+import { Request as ExpressRequest } from 'express';
+import { PassportSubscriptionContext, PassportContext } from 'graphql-passport';
+import { MyUserObject } = 'some_user_file';
+
+export interface MyContext extends PassportContext<MyUserObject, ExpressRequest>{
+  dataSources: {
+    myOtherAPI: { ... }
+  }
+}
+
+export interface MySubscriptionContext extends PassportSubscriptionContext<MyUserObject, ExpressRequest>{
+  dataSources: {
+    myOtherAPI: { ... }
   }
 }
 ```
 
 In your GraphQL functions you can the reference the passport context using:
 
 ```ts
+import { MyContext } from 'somename';
+
 const something = async (
   parent: unknown,
   data: SomeData,
-  context: PassportRequest
+  context: MyContext
 ) => {
-  const username = context.user?.name || "guest";
+  const username = context.getUser()?.name || "guest";
```

Comment:
Is this from the [proposal for optional chaining](https://github.com/tc39/proposal-optional-chaining)? If so, this is not standard JS and will confuse people.

Also, `logger` is not defined. Please remove it

Reply:
Optional chaining is standard as of 3.7 TypeScript (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html). Since this is a TypeScript specific section I don't think it really benefits from `(context.getUser() && context.getUser().name || "guest")` - that is just pure ugly :grimacing: - We could perhaps go all in with `??`. If you insist I can change it though.

Reply:
I will reference this in the new comment about the documentation as well.